tests/integration/query/inline_array/with_filter_all_test.go

```diff
@@ -303,3 +303,45 @@ func TestQueryInlineNotNullBooleanArray_WithAllFilter_Succeeds(t *testing.T) {
 
 	executeTestCase(t, test)
 }
+
+func TestQueryJSONArray_WithAllFilter_Succeeds(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple JSON array, filtered all of string array",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `type Users {
+					name: String
+					custom: JSON
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Shahzad",
+					"custom": [1, false, "second", {"one": 1}, [1, 2]]
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Fred",
+					"custom": [null, false, "second", {"one": 1}, [1, 2]]
+				}`,
+			},
+			testUtils.Request{
```

Comment:
todo: I'm sorry to ask because writing these is still a real pain, but would you mind adding some introspection tests?  The query tests wont actually validate that the gql types are present  and not broken.

Reply:
I've added a basic introspection test. If there's more you'd like to see let me know.

Reply:
Oh of course - sorry - there are no types for the input params. The test you added is a good one to have in general, but the thing I was worried about is not as relevant here as I initially thought.

It could be good to test that the json input type is working correctly though, have a look at `TestFilterForSimpleSchema`.

Reply:
added another test